In [8]:
import numpy as np
import pandas as pd 
import scipy.integrate as integrate
import matplotlib.pyplot as plt
import scipy 

# Ejercicio: Intervalo de Confianza para la Varianza Poblacional ($\sigma^2$)

**Enunciado:**  
Los siguientes pesos, en decagramos, de 10 paquetes de semillas para césped de cierta marca son:  
`46.4, 46.1, 45.8, 47.0, 46.1, 45.9, 45.8, 46.9, 45.2, 46.0`

1. Encuentre el intervalo de confianza del **95%** para la varianza de todos los paquetes de césped que distribuye la marca.
2. Explique por qué es importante que los pesos se asuman con una distribución normal.

In [2]:
import numpy as np
import scipy.stats as stats

# 1. Definición de los datos muestrales
x_datos = np.array([46.4, 46.1, 45.8, 47.0, 46.1, 45.9, 45.8, 46.9, 45.2, 46.0])

# 2. Parámetros básicos de la muestra
n = len(x_datos)               # Tamaño de la muestra
nu = n - 1                     # Grados de libertad (n - 1)
x_mean = np.mean(x_datos)      # Media muestral
S2 = np.var(x_datos, ddof=1)   # Varianza muestral (ddof=1 para dividir entre n-1)

print(f"Tamaño de la muestra (n): {n}")
print(f"Grados de libertad (ν): {nu}")
print(f"Media muestral (x̄): {x_mean:.4f}")
print(f"Varianza muestral (S²): {S2:.6f}")

Tamaño de la muestra (n): 10
Grados de libertad (ν): 9
Media muestral (x̄): 46.1200
Varianza muestral (S²): 0.286222


In [3]:
# 3. Definición del nivel de confianza y significancia
confianza = 0.95
alpha = 1 - confianza          # α = 0.05
alpha_2 = alpha / 2            # α/2 = 0.025 (cola a cada extremo)

# 4. Obtención de valores críticos de la distribución Chi-cuadrada (χ²)
# stats.chi2.ppf recibe la probabilidad acumulada desde la izquierda
chi2_sup = stats.chi2.ppf(1 - alpha_2, nu)  # χ²_0.025 (deja 2.5% a la derecha)
chi2_inf = stats.chi2.ppf(alpha_2, nu)      # χ²_0.975 (deja 2.5% a la izquierda)

print(f"Valor crítico Chi² superior (χ²_0.025, {nu}): {chi2_sup:.4f}")
print(f"Valor crítico Chi² inferior (χ²_0.975, {nu}): {chi2_inf:.4f}")

Valor crítico Chi² superior (χ²_0.025, 9): 19.0228
Valor crítico Chi² inferior (χ²_0.975, 9): 2.7004


In [5]:
# 5. Cálculo de los límites del intervalo de confianza para σ²
limite_inferior = (nu * S2) / chi2_sup
limite_superior = (nu * S2) / chi2_inf

# 6. Despliegue de resultados
print("RESULTADO FINAL DEL INTERVALO DE CONFIANZA")
print(f"Intervalo del {confianza * 100:.0f}% para la varianza poblacional (σ²):")
print(f"{limite_inferior:.3f} < σ² < {limite_superior:.3f}")

RESULTADO FINAL DEL INTERVALO DE CONFIANZA
Intervalo del 95% para la varianza poblacional (σ²):
0.135 < σ² < 0.954


### Expresión del Intervalo de Confianza para la Varianza ($\sigma^2$)

#### 1. Expresión en Términos Algebraicos
El intervalo de confianza del $(1 - \alpha) \times 100\%$ para la varianza de una población normal está dado por:

$$ \frac{(n-1)S^2}{\chi^2_{\alpha/2, \, \nu}} < \sigma^2 < \frac{(n-1)S^2}{\chi^2_{1-\alpha/2, \, \nu}} $$

Donde:
* $n$: Tamaño de la muestra ($n = 10$)
* $\nu = n - 1$: Grados de libertad ($\nu = 9$)
* $S^2$: Varianza muestral
* $(n-1)S^2 = \sum_{i=1}^{n} (x_i - \bar{x})^2$: Suma de desviaciones cuadráticas
* $\chi^2_{\alpha/2, \, \nu}$: Valor crítico superior de Chi-cuadrada
* $\chi^2_{1-\alpha/2, \, \nu}$: Valor crítico inferior de Chi-cuadrada

---

#### 2. Sustitución Numérica del Ejercicio
Sustituyendo los valores del problema ($\nu = 9$, $(n-1)S^2 = 2.576$, $\chi^2_{0.025, 9} = 19.023$, $\chi^2_{0.975, 9} = 2.700$):

$$ \frac{2.576}{19.023} < \sigma^2 < \frac{2.576}{2.700} $$

---

#### 3. Resultado Final
$$ 0.135 < \sigma^2 < 0.953 $$